services/src/main/java/org/keycloak/protocol/saml/installation/KeycloakSamlSubsystemCliInstallation.java

```diff
@@ -0,0 +1,170 @@
+                .append(", \\\n        signResponse=")
+                .append(Boolean.toString(samlClient.requiresClientSignature()))
+                .append(", \\\n        validateRequestSignature=")
+                .append(Boolean.toString(samlClient.requiresRealmSignature()))
+                .append(", \\\n        validateResponseSignature=")
+                .append(Boolean.toString(samlClient.requiresRealmSignature()))
+                .append(", \\\n        requestBinding=POST, \\\n        responseBinding=POST, \\\n        postBindingUrl=")
+                .append(bindingUrl)
+                .append(", \\\n        redirectBindingUrl=")
+                .append(bindingUrl)
+                .append("} \\\n)\n\n");
+
+        if (samlClient.requiresClientSignature()) {
+            builder.append("/subsystem=keycloak-saml/secure-deployment=YOUR-WAR.war/SP=")
+                    .append(quote(entityId))
+                    .append("/IDP=idp/:write-attribute(name=signatureAlgorithm,value=")
+                    .append(samlClient.getSignatureAlgorithm())
+                    .append(")\n\n");
+            if (samlClient.getCanonicalizationMethod() != null) {
+                builder.append("/subsystem=keycloak-saml/secure-deployment=YOUR-WAR.war/SP=")
+                        .append(quote(entityId))
+                        .append("/IDP=idp/:write-attribute(name=signatureCanonicalizationMethod,value=")
+                        .append(samlClient.getCanonicalizationMethod())
+                        .append(")\n");
+            }
+        }
+        
+        return Response.ok(builder.toString(), MediaType.TEXT_PLAIN_TYPE).build();
+    }
+
+    @Override
+    public String getProtocol() {
+        return SamlProtocol.LOGIN_PROTOCOL;
+    }
+
+    @Override
+    public String getDisplayType() {
+        return "Keycloak SAML JBoss Subsystem CLI";
+    }
+
+    @Override
+    public String getHelpText() {
+        return "CLI script you must edit and apply to your client app server. This type of configuration is useful when you can't or don't want to crack open your WAR file.";
+    }
+
+    @Override
+    public String getFilename() {
+        return "keycloak-saml-subsystem.cli";
+    }
+
+    @Override
+    public String getMediaType() {
+        return MediaType.TEXT_PLAIN;
+    }
+
+    @Override
+    public boolean isDownloadOnly() {
+        return false;
+    }
+
+    @Override
+    public void close() {
+
+    }
+
+    @Override
+    public ClientInstallationProvider create(KeycloakSession session) {
+        return this;
+    }
+
+    @Override
+    public void init(Config.Scope config) {
+
+    }
+
+    @Override
+    public void postInit(KeycloakSessionFactory factory) {
+
+    }
+
+    @Override
+    public String getId() {
+        return "keycloak-saml-subsystem-cli";
+    }
+
+    private String quote(String value) {
+        return "\"" + value + "\"";
```

Comment:
The `value` has to be quoted per https://access.redhat.com/documentation/en-us/red_hat_jboss_enterprise_application_platform/7.2/html-single/management_cli_guide/index#cli_special_characters